netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -698,6 +698,9 @@ protected void userEventTriggered0(ChannelHandlerContext ctx, Object evt) throws
             // Successfully negotiated the protocol.
             logSslEngineDetails(Level.FINER, ctx, "TLS negotiation succeeded.", null);
             propagateTlsComplete(ctx, handler.engine().getSession());
+            if (handshakeCompleteRunnable.isPresent()) {
```

Comment:
Is there a reason not to have a single copy of the code within the `evt instanceof SslHandshakeCompletionEvent` block?